gateway/src/apicast/configuration_loader.lua

```diff
@@ -196,33 +196,30 @@ function lazy.init(configuration)
   configuration.configured = true
 end
 
+function lazy.load_config(configuration, host)
+    local config = _M.load(host)
+    if not config then
+      ngx.log(ngx.WARN, 'failed to get config for host: ', host)
+    end
+    _M.configure(configuration, config)
+end
+
 function lazy.rewrite(configuration, host)
   if not host then
     return nil, 'missing host'
   end
 
-  local sema = synchronization:acquire(host)
-
   if ttl() == 0 then
     configuration = configuration_store.new(configuration.cache_size)
   end
 
-  local ok, err = sema:wait(15)
-
-  if ok and not _M.configured(configuration, host) then
-    ngx.log(ngx.INFO, 'lazy loading configuration for: ', host)
-    local config = _M.load(host)
-    if not config then
-      ngx.log(ngx.WARN, 'failed to get config for host: ', host)
-    end
-    _M.configure(configuration, config)
+  if _M.configured(configuration, host) then
+    return configuration
   end
 
-  if ok then
-    synchronization:release(host)
-    sema:post()
-  else
-    ngx.log(ngx.WARN, 'failed to acquire lock to lazy load: ', host, ' error: ', err)
+  local ret, result = synchronization:run(host, 15, lazy.load_config, configuration, host)
```

Comment:
Can we define a constant with this 15? I know it's not a problem introduced in this PR, but looks like a good opportunity to fix this.